### PR TITLE
Refactor `cmd/install.py` for better readability

### PR DESCRIPTION
**Pull request by alalazo**
Modifications:
- [x] Split long functions into multiple logical units
- [x] Renamed a few variables to better convey their properties (e.g. `specs` -> `concrete_specs`)
- [x] Removed dead code

### CODE REVIEW

File: lib/spack/spack/cmd/install.py [hunk 1]
```diff
@@ -27,43 +27,28 @@
...
-        }
-    )
-
-    if hasattr(args, "setup"):
-        setups = set()
-        for arglist_s in args.setup:
-            for arg in [x.strip() for x in arglist_s.split(",")]:
-                setups.add(arg)
-        kwargs["setup"] = setups
-        tty.msg("Setup={0}".format(kwargs["setup"]))
```
Comment (alalazo) [inline lib/spack/spack/cmd/install.py:66]: For the record: I completely removed this code during the refactor, since it doesn't seem to be called anywhere

**Comment by alalazo** [comment]
@spackbot run pipeline

File: lib/spack/spack/cmd/install.py [hunk 2]
```diff
@@ -27,43 +27,28 @@
...
+        "keep_stage": args.keep_stage,
+        "restage": not args.dont_restage,
+        "install_source": args.install_source,
+        "verbose": args.verbose or args.install_verbose,
+        "fake": args.fake,
+        "dirty": args.dirty,
+        "use_cache": args.use_cache,
+        "cache_only": args.cache_only,
+        "include_build_deps": args.include_build_deps,
+        "explicit": True,  # Always true for install command
```
Comment (haampie) [inline lib/spack/spack/cmd/install.py]: FWIW, this comment is not 100% true anymore: `spack -e my_env install --no-add --only-concrete /hash` is `explicit` if and only if `/hash` is a root spec, but it's a sensible default...

Comment (alalazo) [reply lib/spack/spack/cmd/install.py]: Should be fixed

**Comment by alalazo** [comment]
Thanks for the offline comments and review @haampie !

**Review by haampie — APPROVED** [review]

**Pull request merged**